source/slang/legalize-types.cpp

```diff
@@ -916,7 +916,7 @@ LegalType legalizeType(
             }
 
             legalType = builder.getResult();
-            context->mapDeclRefToLegalType.Add(declRef, legalType);
+            context->mapDeclRefToLegalType.AddIfNotExists(declRef, legalType);
```

Comment:
I assume this is a workaround for the crashing issue. It still seems weird that we'd hit this case at all.